app/src/main/java/com/example/playlistmaker/ui/SearchActivity.kt

```diff
@@ -30,15 +33,16 @@ class SearchActivity : AppCompatActivity() {
 
     private var inputSearchText: String = DEFAULT_TEXT
 
-    private val itunesBaseUrl = "https://itunes.apple.com"
-    private val retrofit = Retrofit.Builder().baseUrl(itunesBaseUrl).addConverterFactory(
-        GsonConverterFactory.create()
-    ).build()
+    private var retrofit = RetrofitInit()
+    private val itunesService = retrofit.getService()
 
-    private val itunesService = retrofit.create(ItunesApi::class.java)
+    private val tracksSearchDatabase = TracksStorage.getInstance()
+    private val tracksSearchRepository = TracksSearchRepository(itunesService, tracksSearchDatabase)
```

Comment:
Если хочешь улучшить производительность, то можно инициализировать эту сущность только тогда когда к ней обратятся, и так с остальными
```suggestion
private val tracksSearchRepository by lazy {
        TracksSearchRepository(itunesService, tracksSearchDatabase)
    }
```